quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/KcEnvConfigSource.java

```diff
@@ -37,9 +39,22 @@ private static Map<String, String> buildProperties() {
 
         for (Map.Entry<String, String> entry : System.getenv().entrySet()) {
             String key = entry.getKey();
+            String value = entry.getValue();
 
             if (key.startsWith(kcPrefix)) {
-                properties.put(getMappedPropertyName(key), entry.getValue());
+                properties.put(key, value);
```

Comment:
do we really need to add up to 3 times to properties? so first the raw key/value, then when mapper exists and `to` is defined, then when `from` is defined, again?

Will give the PR a deeper look tomorrow, but would be nice if you could elaborate a bit here, can't think of a situation atm where at least the `getFrom` wouldn't be doubled, but may be I miss something :)

Reply:
ok, to answer this myself.. debugged it. There are 3 different keys added, see picture of debug output.
<img width="381" alt="Screenshot 2022-06-22 at 11 17 54" src="https://user-images.githubusercontent.com/89905860/174992857-83b60c35-af43-45ff-99a8-867a54d222ea.png">

Subsequent question: Would it be worth to not add the `KC_DB_...` key to the properties array, but directly replacing it, by e.g. parsing it first to the desired `kc.<option-name>` format from the format of the env var and removing the from part? Thinking of not polluting the properties, which makes traversing them a bit less performant, i guess. but maybe that's too much. wdyt @pedroigor

Reply:
There is no performance impact when traversing because we are using a map to look up options.

And it does not work because we have the SPI options which are not mapped at all in our configuration layer. The `ConfigurationTest` is covering this area.

Reply:
aaah right, sorry I forgot about the other options. then fine for me :)